libs/content/workspace/src/task_manager.rs

```diff
@@ -343,55 +490,91 @@ impl TaskManager {
 
-                    let completed_load = CompletedLoad {
-                        request: in_progress_load.request,
-                        content_result,
-                        timing: CompletedTiming::new(in_progress_load.timing),
-                    };
-                    let in_progress_time = completed_load
-                        .timing
-                        .started_at
-                        .duration_since(completed_load.timing.queued_at);
-                    if in_progress_time > Duration::from_secs(1) {
-                        warn!("Load of file {} took {}s", request.id, in_progress_time.as_secs());
+                    let timing = CompletedTiming::new(in_progress_load.timing);
+                    let in_progress_time = timing.completed_at.duration_since(timing.started_at);
+                    if let Err(err) = &content_result {
+                        error!(
+                            "load of file {} failed ({}ms): {:?}",
+                            request.id,
+                            in_progress_time.as_millis(),
+                            err
+                        );
+                    } else if in_progress_time > Duration::from_secs(1) {
+                        warn!("loaded file {} ({}ms)", request.id, in_progress_time.as_millis());
+                    } else {
+                        debug!("loaded file {} ({}ms)", request.id, in_progress_time.as_millis());
                     }
+
+                    let completed_load =
+                        CompletedLoad { request: in_progress_load.request, content_result, timing };
                     tasks.completed_loads.push(completed_load);
                 }
 
-                self_clone.check_launch(); // task completion may trigger launch of queued task
-                self_clone.ctx.request_repaint(); // task completion affects UI
+                self_clone.ctx.request_repaint();
             });
         }
 
         for queued_save in saves_to_launch.into_values() {
-            // content cloned; one copy sent to disk and other retained in UI to represent on-disk version for merge
-            // first step to alleviate: https://github.com/lockbook/lockbook/issues/3241
             let request = queued_save.request.clone();
             let in_progress_save = InProgressSave::new(queued_save);
+            let content = {
+                let mut result = None;
+                for tab in tabs {
+                    if tab.id == request.id {
+                        let start = Instant::now();
+                        result = tab.content.clone();
+                        let time = Instant::now().duration_since(start);
+                        if time > Duration::from_millis(100) {
+                            warn!(
+                                "spent {}ms on UI thread cloning content for file {} for background thread to save it",
+                                request.id,
+                                time.as_millis()
```

Comment:
I believe duration has put some effort into readability by impl'ing a display, could be wrong